nodebuilder/tests/api_test.go

```diff
@@ -58,6 +58,10 @@ func TestNodeModule(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, info.APIVersion, node.APIVersion)
 
+	ready, err := client.Node.Ready(ctx)
+	require.NoError(t, err)
+	require.True(t, ready)
+
```

Comment:
Does it block when it's not ready?

Reply:
For context, see the issue - rn nodes are using some arbitrary endpoint for this like network head

Reply:
I understand the purpose. My question is what happens when its not ready

Reply:
If its not ready the rpc doesnt connect, so a connection error is returned by whatever library you are using to talk to the RPC. I get that its a bit silly, but we cannot provide an endpoint before the node is ready anyways

Reply:
So that could be any endpoint. I don't mind having a special endpoint, though. We could also think of any other checks we can add to be sure the node is ready. Like p2p networking is ready and we have some connections

Reply:
Preferably all services started would be nice.

Reply:
Discussed with Rene yesterday - it is a bigger issue but one thing we can ensure is that RPC is only started after all other service's lifecycles have begun. This should be enough for now, and implementing something more robust would require implementing readiness into the services themselves to query. I will add it to the PR today

Reply:
So RPC server has already started the last. Could you also check if we connect to bootstrappers before RPC?

Reply:
I can confirm. Cannot connect to node until bootstrapper connection is established as it is part of successful startup, and RPC calls fail before startup is finished

Reply:
Cool, as the last step let's comment out inside Ready that it's in fact awaiting for Services and connections

Reply:
personally always like having a /health(z) that is a simple "is it started" for simple up/down health checks (which this /ready check satisfies and i think the real ask so things like container orchestrators can use just to start serving traffic to the instance) , then a separate /status or something which can validates each internal type of connection/required service, so /status returns a check of all, then /status/{p2p,storage...} might exist for checking each individually. Readiness, vs liveness separately.

Reply:
In my practice, what you call /healthz is /readyz) then and what you call /status is /healthz. Also, it makes more sense to call simple "is it started" as is /readyz